packages/number-input/src/number-input.tsx

```diff
@@ -152,11 +155,18 @@ export interface NumberInputFieldProps extends HTMLChakraProps<"input"> {}
  */
 export const NumberInputField = forwardRef<NumberInputFieldProps, "input">(
   (props, ref) => {
-    const { getInputProps } = useNumberInputContext()
+    const { hasSteppers, getInputProps } = useNumberInputContext()
 
     const input = getInputProps(props, ref)
     const styles = useStyles()
 
+    const paddingRight = useToken("space", styles.field.px?.toString() ?? 0)
+    const stepperWidth = styles.stepperGroup.width
+
+    if (hasSteppers) {
+      styles.field.paddingRight = `calc(${paddingRight} + ${stepperWidth})`
+    }
```

Comment:
If the user uses `extendTheme` and changes some of these values, this might produce wrong behavior.

You can skip checking if `hasSteppers` and just add the padding by default. It's very unlikely that the users will use number-input without steppers.

Reply:
I'd move most of these styles to the theme as well. Also, note to use `paddingEnd` instead of `paddingRight` to make the component work for RTL by default.

Reply:
That makes sense as well. I didn't thought of the RTL honestly. Thanks for this @segunadebayo!